kernel/term.mli

```diff
@@ -42,13 +42,18 @@ val subterm : term -> position -> term
 
 (** {2 Contexts} *)
 
+type 'a context = (loc * ident * 'a) list
 (** Abstract context *)
-type 'a context = 'a list
 
-(** context of rules after they have been parsed *)
-type untyped_context = (loc * ident) context
+type part_typed_context = term option context
+(** Rule context after they have been parsed *)
+
+type typed_context = term context
+(** Rule context after type checking assigns a type to the variables *)
+
+val pp_untyped_context    : 'a context         printer
```

Comment:
Why three printer for context ? We should have only a printer for the type of elements in the context.